Documentation: Changing "combinatorial" to "combinational" (#1332)
amaranth-lang/amaranth: docs/contrib.rst

* For properties, phrase the short description (first line of docstring) like ``Value of thing.``, i.e. as a declarative sentence.
* When documenting signatures of interfaces, as well as components, use the (non-standard) `Members` section to document their interface members, and only that section; do not document them in an `Attributes` section.
* If an anchor for a section is needed, namespace it, e.g. the ``.. _lang-assignable:`` anchor is a part of the ``lang`` namespace. Anchor names are global.

> * It is preferred to use the term "combinational" rather than "combinatorial" to refer to non-sequential logic. See `Issue #1301`_ for more details.

Reviewer: ```suggestion
* To refer to non-sequential logic, use the term "combinational" over "combinatorial".
```

(A style guide's purpose is to set the standard. It is stronger than a mere _preference_--you are required to use this term when contributing--and it's also not particularly important what the reasoning is, just like for all other items here.)
